pkg/autoscaler/multiscaler_test.go

```diff
@@ -128,6 +130,69 @@ func TestMultiScalerScaling(t *testing.T) {
 	}
 }
 
+func TestMultiScalerTickUpdate(t *testing.T) {
+	ctx := context.Background()
+	ms, stopCh, statCh, uniScaler := createMultiScaler(t)
+	defer close(stopCh)
+	defer close(statCh)
+
+	decider := newDecider()
+	decider.Spec.TickInterval = 10 * time.Second
+	uniScaler.setScaleResult(1, true)
+
+	// Before it exists, we should get a NotFound.
+	m, err := ms.Get(ctx, decider.Namespace, decider.Name)
+	if !apierrors.IsNotFound(err) {
+		t.Errorf("Get() = (%v, %v), want not found error", m, err)
+	}
+
+	_, err = ms.Create(ctx, decider)
+	if err != nil {
+		t.Errorf("Create() = %v", err)
+	}
+
+	// ticker will execute the scale count for 50ms with interval 1ms to ensure nothing happened.
+	timeoutTicker := time.NewTicker(50 * time.Millisecond)
+	intervalTicker := time.NewTicker(time.Millisecond)
+	defer timeoutTicker.Stop()
+	defer intervalTicker.Stop()
+	errChan := make(chan error)
+
+	go func(chan error) {
+		for {
+			select {
+			case <-intervalTicker.C:
+				// Expected count to be 0 as the tick interval is 10s and no autoscaling calculation should be triggered
+				if count := uniScaler.getScaleCount(); count != 0 {
+					errChan <- fmt.Errorf("Want count= 0, got count=%d", count)
+				}
+			}
+		}
+	}(errChan)
+
+	select {
+	case <-errChan:
+	case <-timeoutTicker.C:
+		intervalTicker.Stop()
+	}
```

Comment:
I think this is a bit over the top. Does this suffice:

```golang
time.Sleep(50 * time.Millisecond)
if count := uniScaler.getScaleCount(); count != 0 {
  t.Errorf(...)
}
```

? I think you only want/need to check here that the ticker hasn't actually ticked in this interval. No need to check that multiple times, that's encoded in the `getScaleCount` anyway. Or am I missing something?

Reply:
Yes you are right. I have updated the test